pkg/types/gcp/platform.go

```diff
@@ -49,6 +60,13 @@ type Platform struct {
 	// feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags
 	// featureGate enabled or TechPreviewNoUpgrade featureSet to configure tags.
 	UserTags []UserTag `json:"userTags,omitempty"`
+
+	// UserProvisionedDNS indicates if the customer is providing their own DNS solution in place of the default
+	// provisioned by the Installer.
+	// +kubebuilder:default:="Disabled"
+	// +default="Disabled"
+	// +kubebuilder:validation:Enum="Enabled";"Disabled"
+	UserProvisionedDNS UserProvisionedDNS `json:"userProvisionedDNS,omitempty"`
```

Comment:
This and the above enums seem like they would be better placed in the install-config. What do you think?

Reply:
Yes, these enums could be moved to a common location as we start adding platforms.

Reply:
@sadasu and @jhixson74 the enums can be made common but not the work in to the install config. There was push back originally for adding it for all platforms it was stated that it would be confusing if users could select this option when platforms dont have that ability.